Take boolean attributes into account

ryota-ka opened the pull request:
Updated `modules/props.js` so that it will take [boolean attributes](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes) into account.
(`false` values for such attributes shouldn't be interpreted as `"false"`)

acstll commented on modules/props.js:
```diff
@@ -49,6 +81,13 @@ module.exports = function propsModule (vnode, attributes) {
-    attributes.set(key.toLowerCase(), escape(value))
+    var lkey = key.toLowerCase()
+    if (~booleanAttributes.indexOf(lkey)) {
+      if (value) { // set attr only when true
```
if you really mean `true`, then I would check with `value === true`, otherwise I would say 'truthy' in the comment

```suggestion
      if (value) { // set attr only when truthy
```

ryota-ka replied:
Updated!

acstll commented:
Thank you for the PR.

Do you think you could add a test for this change?

ryota-ka commented:
@acstll Thank you for your quick response.

I've added a test case for this change.
Also, I've changed `const` to `var` in 00c54761598bcde84a6d1d9f9a80986aae777cd8.

If it looks fine, please let me know before merging. I'll squash my commits.

acstll commented:
Looks good to me!

I'll ping @paldepind to double check before releasing this. I saw he was active on this project recently, and I haven't touched this in a very long time.

This is a bug fix but I'd do a major (6.0.0) since I believe this could break people's code, no?

ryota-ka commented:
I've squashed the `fixup!` commit and force-pushed.

@paldepind Please feel free to merge this if you feel it's okay 🙏

ryota-ka commented:
@paldepind Pinging for your review (Sorry for interruption :bow:)

ryota-ka commented:
@acstll It's been four months.
Shall we wait for @paldepind's review any more?

Personally I'm also :+1: for making a major version bump.

acstll commented:
Hey, time flies. Sorry you had to wait this long. I think we can move forward. I'll merge and cut a major release this week.

acstll commented:
It's done @ryota-ka I hope everything's OK, haven't done this in a while! Thanks again for your patience.

ryota-ka commented:
@acstll I also appreciate your review and releasing the new version! I'll enjoy it!
